sky/shell/mac/tracing_controller.mm

```diff
@@ -2,119 +2,130 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#import "tracing_controller.h"
-#include <string>
+#include "base/logging.h"
 #include "base/macros.h"
 #include "base/trace_event/trace_config.h"
 #include "base/trace_event/trace_event.h"
-
-@interface TracingController ()
-@property(nonatomic, retain) NSFileHandle* currentFileHandle;
-@end
+#include "sky/shell/mac/tracing_controller.h"
+#include "sky/shell/shell.h"
+#include <Foundation/Foundation.h>
+#include <string>
 
 namespace sky {
 namespace shell {
 
-const char kStart[] = "{\"traceEvents\":[";
-const char kEnd[] = "]}";
+static const std::string kBaseTraceStart("{\"traceEvents\":[");
+static const std::string kBaseTraceEnd("]}");
+static const std::string kSentinel("\0", 1);
```

Comment:
We can't have static std::strings because that creates static initializers, which slows down startup.  We should use const char kXXX[] instead.